tests/test_magic.rs

```diff
@@ -0,0 +1,105 @@
+use deku::prelude::*;
+use hexlit::hex;
+use rstest::rstest;
+use std::convert::{TryFrom, TryInto};
+
+pub mod samples {
```

Comment:
I got rid of this `samples` pattern during a recent refactor, you can place the structs/enums inside the test functions :) Can check `test_cond.rs` for an example

Reply:
Also used `TestStruct` or `TestEnum` inside the test functions for consistency

Reply:
Oh, yep, I missed that that refactor had removed the `samples` pattern